outputs.tf

```
@@ -21,4 +21,9 @@ output "mcd_agent_security_group_name" {
 output "mcd_agent_storage_bucket_arn" {
   value       = aws_s3_bucket.mcd_agent_store.arn
   description = "Storage bucket ARN."
-}
\ No newline at end of file
+}
+
+output "mcd_agent_execution_role" {
+  value = aws_iam_role.mcd_agent_service_execution_role
+  description = "The execution role for the in the customer account Lambda"
+}
```

I believe you need to specify a specific field from the resource. For instance: 

```suggestion
output "mcd_agent_execution_role_arn" {
  value = aws_iam_role.mcd_agent_service_execution_role.arn
  description = "The execution role ARN for the agent lambda."
}
```

---

You don't have to specify the specific attribute. Since some downstream consumers may want to reference the arn, while others may just want to reference the role name, I left the output at this highest level supported resource so the desired attribute could be selected per each customer

---

Gotcha. I thought the intent was to display only and not use it downstream, similar to how other outputs are currently handled.